### PR TITLE
some docs fixes (replacing old links with new one )

### PR DESCRIPTION
**Pull request by Sauravroy34**
for #7761

### CODE REVIEW

File: docs/dev_guide/contents/newcomers.rst [hunk 1]
```diff
@@ -127,9 +127,9 @@ Setting up a development environment
 
 The instructions in this following section are based upon three resources:
 
-* `Astropy Dev Workflow <https://docs.astropy.org/en/latest/development/workflow/development_workflow.html>`_
-* `Astropy Dev environment <https://docs.astropy.org/en/latest/development/workflow/get_devel_version.html#get-devel>`_
-* `Astropy Pull Request Example <https://docs.astropy.org/en/latest/development/workflow/git_edit_workflow_examples.html#astropy-fix-example>`_
+* `Astropy Dev Workflow <https://kbarbary-astropy.readthedocs.io/en/latest/development/workflow/development_workflow.html>`_
```
Comment (ayshih) [inline docs/dev_guide/contents/newcomers.rst:130]:
```suggestion
```

File: docs/dev_guide/contents/newcomers.rst [hunk 2]
```diff
@@ -127,9 +127,9 @@ Setting up a development environment
...
+* `Astropy Dev environment <https://docs.astropy.org/en/latest/development/quickstart.html>`_
```
Comment (ayshih) [inline docs/dev_guide/contents/newcomers.rst:131]:
```suggestion
* `Astropy Contributing Quickstart <https://docs.astropy.org/en/latest/development/quickstart.html>`_
```

File: docs/dev_guide/contents/newcomers.rst [hunk 3]
```diff
@@ -127,9 +127,9 @@ Setting up a development environment
...
+* `Astropy Pull Request Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`_
```
Comment (ayshih) [inline docs/dev_guide/contents/newcomers.rst:132]:
```suggestion
* `Astropy Contribution Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`_
```

File: docs/dev_guide/contents/newcomers.rst (hunk 1 above)
Comment (ayshih) [reply docs/dev_guide/contents/newcomers.rst:130]: This page doesn't exist anymore.  We definitely shouldn't link to someone else's rendering of super-old Astropy docs.

**Review by ayshih — CHANGES_REQUESTED** [review]
In addition, I'm of the opinion that we should outright remove the references to Anaconda.

File: docs/dev_guide/contents/newcomers.rst [hunk 4]
```diff
@@ -127,9 +127,8 @@ Setting up a development environment
 
 The instructions in this following section are based upon three resources:
 
-* `Astropy Dev Workflow <https://docs.astropy.org/en/latest/development/workflow/development_workflow.html>`_
-* `Astropy Dev environment <https://docs.astropy.org/en/latest/development/workflow/get_devel_version.html#get-devel>`_
-* `Astropy Pull Request Example <https://docs.astropy.org/en/latest/development/workflow/git_edit_workflow_examples.html#astropy-fix-example>`_
+* `Astropy Contributing Quickstart <https://docs.astropy.org/en/latest/development/quickstart.html>`_
+* `Astropy Contribution Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`_
```
Comment (nabobalis) [inline docs/dev_guide/contents/newcomers.rst:131]:
```suggestion
* `Astropy Contributing Quickstart <https://docs.astropy.org/en/latest/development/quickstart.html>`__
* `Astropy Contribution Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`__
```

File: docs/dev_guide/contents/newcomers.rst [hunk 5]
```diff
@@ -185,7 +184,6 @@ Ideally, when you start making changes you want to create a git branch:
 You can change ``my_fix`` to anything you prefer.
 If you get stuck or want help, just `ask here`_!
 
-.. _Anaconda: https://www.anaconda.com/distribution/
 .. _miniconda: https://conda.io/en/latest/miniconda.html
```
Comment (nabobalis) [inline docs/dev_guide/contents/newcomers.rst:187]:
```suggestion
.. _miniforge: hhttps://github.com/conda-forge/miniforge?tab=readme-ov-file#download
```

File: docs/dev_guide/contents/newcomers.rst [hunk 6]
```diff
@@ -127,14 +127,13 @@ Setting up a development environment
...
-* `Astropy Dev environment <https://docs.astropy.org/en/latest/development/workflow/get_devel_version.html#get-devel>`_
-* `Astropy Pull Request Example <https://docs.astropy.org/en/latest/development/workflow/git_edit_workflow_examples.html#astropy-fix-example>`_
+* `Astropy Contributing Quickstart <https://docs.astropy.org/en/latest/development/quickstart.html>`__
+* `Astropy Contribution Example <https://docs.astropy.org/en/latest/development/git_edit_workflow_examples.html>`__
 
 **We strongly recommend that you read these links.**
 These links are more in-depth than this guide but you will need to replace ``astropy`` with ``sunpy``.
 
-In order to start coding you will need a local Python environment and we would recommend using `Anaconda`_ or `miniconda`_ (shortened to conda from here on).
+In order to start coding you will need a local Python environment and we would recommend using Anaconda or `miniconda`_ (shortened to conda from here on).
```
Comment (nabobalis) [inline docs/dev_guide/contents/newcomers.rst:136]:
```suggestion
In order to start coding you will need a local Python environment and we would recommend using `miniforge`_ (shortened to conda from here on).
```

**Review by ayshih — APPROVED** [review]
There's one more reference to Anaconda to remove on line 175 of `newcomers.rst`, otherwise, looks good to me

**Comment by nabobalis** [comment]
Thanks for the PR @Sauravroy34

**Pull request merged**